dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionMetadata.cs

```diff
@@ -19,12 +20,16 @@ public sealed class KernelFunctionMetadata
     private IReadOnlyList<KernelParameterMetadata> _parameters = [];
     /// <summary>The function's return parameter.</summary>
     private KernelReturnParameterMetadata? _returnParameter;
+    /// <summary>A static empty dictionary to default to when none is provided.</summary>
+    private static readonly Dictionary<string, object?> s_emptyDictionary = new();
 
     /// <summary>Initializes the <see cref="KernelFunctionMetadata"/> for a function with the specified name.</summary>
     /// <param name="name">The name of the function.</param>
+    /// <param name="propertyBag">Optional metadata in addition to the named properties already available on this class.</param>
     /// <exception cref="ArgumentNullException">The <paramref name="name"/> was null.</exception>
     /// <exception cref="ArgumentException">An invalid name was supplied.</exception>
-    public KernelFunctionMetadata(string name)
+    public KernelFunctionMetadata(string name, ReadOnlyDictionary<string, object?>? propertyBag = null)
```

Comment:
Is there an option to add new constructor overload with `propertyBag` parameter instead of modifying existing constructor? In this case there will be no need in compatibility suppression file.

Also, should be name new parameter as `metadata` instead of `propertyBag`? I think `metadata` should work better, based on description of this parameter.

Reply:
Makes sense!

Reply:
How about `additionalMetadata` or `extensionMetadata`